packages/@aws-cdk/aws-elasticloadbalancingv2/lib/nlb/network-listener.ts

```diff
@@ -86,12 +87,52 @@ export interface NetworkListenerProps extends BaseNetworkListenerProps {
   readonly loadBalancer: INetworkLoadBalancer;
 }
 
+/**
+ * Options for looking up a network listener.
+ */
+export interface NetworkListenerLookupOptions extends BaseListenerLookupOptions {
+  /**
+   * Protocol of the listener port
+   * @default - listener is not filtered by protocol
+   */
+  readonly listenerProtocol?: Protocol;
+}
+
 /**
  * Define a Network Listener
  *
  * @resource AWS::ElasticLoadBalancingV2::Listener
  */
 export class NetworkListener extends BaseListener implements INetworkListener {
+  /**
+   * Looks up a network listener
+   */
+  public static fromLookup(scope: Construct, id: string, options: NetworkListenerLookupOptions): INetworkListener {
+    let listenerProtocol: cxschema.LoadBalancerListenerProtocol | undefined = undefined;
+    if (options.listenerProtocol) {
+      validateNetworkProtocol(options.listenerProtocol);
+
+      switch (options.listenerProtocol) {
+        case Protocol.TCP: listenerProtocol = cxschema.LoadBalancerListenerProtocol.TCP; break;
+        case Protocol.UDP: listenerProtocol = cxschema.LoadBalancerListenerProtocol.UDP; break;
+        case Protocol.TCP_UDP: listenerProtocol = cxschema.LoadBalancerListenerProtocol.TCP_UDP; break;
+        case Protocol.TLS: listenerProtocol = cxschema.LoadBalancerListenerProtocol.TLS; break;
+      }
+    }
+
+    const props = BaseListener._queryContextProvider(scope, {
+      userOptions: options,
+      listenerProtocol: listenerProtocol,
+      loadBalancerType: cxschema.LoadBalancerType.NETWORK,
+    });
+
+    class LookedUp extends Resource implements INetworkListener {
+      public listenerArn = props.listenerArn;
+    }
```

Comment:
This seems like a case where following a pattern and having commonality between the ALB and NLB interface yields a sub-optimal result. `NetworkListenerLookupOptions` extends `BaseListenerLookupOptions`, which includes `listenerArn` as one of the filtering options, but there's no case here for why someone would use `fromLookup` with `listenerArn` over `fromNetworkListenerArn`.

Reply:
@njlynch Makes sense. Is your suggestion then to entirely separate the interfaces for `ApplicationListener.fromLookup()` and `NetworkListener.fromLookup()`? Or to push `BaseListenerLookupOptions.listenerArn` down to `ApplicationListenerLookupOptions.listenerArn`? Or...?

Reply:
Well, as a bit of devil's advocate, what use case does adding the lookup by ARN enable? What specific functionality for `ApplicationListener.fromLookup()` is benefitted from `ApplicationListener.fromLookup({ listenerArn: 'foo' })` vs `ApplicationListener.fromApplicationListenerAttributes()` with the ARN?

Reply:
@njlynch Fair enough comments. But I had in mind that if the user has the ARN of the listener, it's nice to have CDK get the security groups of the ALB by lookup as well. For instance, the user is probably going to need the ALB security groups to let the ALB traffic into an ECS/Fargate task.

They can certainly use `ApplicationListener.fromApplicationListenerAttributes()` with the ids of all involved resources, but then they'll need to provide the security groups of the load balancer as well. So, the improvement of lookup by ARN is primarily a convenience over needing to specify the security group.

If NLB's listener's interface had `ec2.IConnectable`, I'd probably say the same for it, but only `IApplicationListener` extends the interface. So ALB and NLB listeners are definitely a little incongruous in that respect.

Reply:
Fair point. I can see the benefit of having the security groups automatically loaded. Perhaps we move `listenerArn` up to `ApplicationListenerLookupOptions` then. I'm also going to ask one of my colleagues to take a look as this PR since it's so big, and see if they have an opinion.

Reply:
I like moving `listenerArn` to `ApplicationListenerLookupOptions` if providing the ARN doesn't make sense for Network Listeners.

We can always move it back down, but we can't remove it after it's added to `NetworkListenerLookupOptions` 🙂.